src/Signup.jsx

```diff
@@ -1,3 +1,185 @@
-const Signup = () => <h2>This is the signup page!</h2>;
+import { useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+
+import Neopost from './assets/icons/neopost.svg';
+import BackgroundVector from './assets/background-vector.png';
+import PasswordInput from './PasswordInput';
+import TextInput from './TextInput';
+import toast from 'react-hot-toast';
+
+import './App.scss';
+
+const onError = () => {
+  toast.error('Please fix the errors in the form.');
+};
+
+const Signup = () => {
+  const {
+    register,
+    handleSubmit,
+    watch,
+    setError,
+    resetField,
+    formState: { errors }
+  } = useForm({ mode: 'onChange' });
+
+  const allFieldsFilled = watch('email') && watch('name') && watch('password') && watch('password_confirmation');
+
+  const password = watch('password');
+  const apiUrl = import.meta.env.VITE_API_URL;
+  const navigate = useNavigate();
+
+  const onSubmit = async(data) => {
+    try {
+      const response = await fetch(`${apiUrl}users`, {
```

Comment:
lets use the apiURL on the .env without the/, it looks better that way here :)
```suggestion
      const response = await fetch(`${apiUrl}/users`, {
```